### ISSUE TITLE
[Feature]: Healthcheck

### ISSUE THREAD

**Issue by mrjvs** [PretendoNetwork/chubby#22]
### Checked Existing

- [X] I have checked the repository for duplicate issues.

### What feature do you want to see added?

A way to healthcheck the service.

### Why do you want to have this feature?

This is useful in infrastructure automations. It allows automation tooling to determine when a service is ready for connections or when a service is in an unrecoverable state in which it needs to restart.

### Any other details to share? (OPTIONAL)

Healthcheck guidelines:
- A healthcheck endpoint can be two types:
  - A HTTP endpoint on `/health` that responds with 200 if healthy and 500 if unhealthy
  - A GRPC API that implements the healthcheck spec: https://github.com/grpc/grpc/blob/master/doc/health-checking.md
- A healthcheck should report unhealthy if it hasn't finished initializing yet:
  - Still loading code: unhealthy
  - Loading files from file system: unhealthy
  - Connecting to a third party service: healthy
- A healthcheck should only report unhealthy if it's truly unrecoverable:
  - Deadlocked?: unhealthy
  - Cant reach db: still healthy
  - Can't read any files from disk: unhealthy
- A good minimal healthcheck would be one that returns healthy if it's done initializing (without counting third party connections)

**Comment by jonbarrow** [comment]
Is there a typo here or am I misunderstanding something? You say both:

> A healthcheck should report unhealthy if it hasn't finished initializing

*and*

> A healthcheck should only report unhealthy if it's truly unrecoverable

Which are mutually exclusive things

**Comment by mrjvs** [comment]
you're right @jonbarrow, I updated the wording a bit. its unhealthy only if it hasnt finished initializing **or** if its in an unrecoverable state.
